File: app/src/main/java/net/north101/android/ghplayertracker/data/PlayedCards.java
package net.north101.android.ghplayertracker.data;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

import java.util.ArrayList;

@Parcel
public class PlayedCards {
    public final ArrayList<Card> pile1;
    public final ArrayList<Card> pile2;
    public boolean shuffled;

    @ParcelConstructor
    public PlayedCards(ArrayList<Card> pile1, ArrayList<Card> pile2, boolean shuffled) {
        this.pile1 = pile1;
        this.pile2 = pile2;
        this.shuffled = shuffled;
    }
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/PlayedCards.kt
package net.north101.android.ghplayertracker.data

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import net.north101.android.ghplayertracker.RecyclerItemCompare

@Parcelize
data class PlayedCards(
        val pile1: ArrayList<Card>,
        val pile2: ArrayList<Card>?,
        var shuffledIndex: Int?
) : Parcelable, RecyclerItemCompare {
    override val compareItemId: String
        get() {
            val id1 = pile1.joinToString(",", "[", "]") { it.id }
            val id2 = pile2?.joinToString(",", "[", "]") { it.id } ?: ""
            return id1 + id2
        }

    fun hasShuffle(): Boolean {
        return pile1.any { it.special == CardSpecial.Shuffle } || pile2?.any { it.special == CardSpecial.Shuffle } ?: false
    }
}
